### ISSUE TITLE
Empty multiple selection HTML is unclear.

### ISSUE THREAD

**Issue by maryokhin** [tomchristie/django-rest-framework#2579]
With a field on a serializer like this:
```python
image_ids = serializers.PrimaryKeyRelatedField(source='images', write_only=True, many=True, queryset=Image.objects.all())
```
I get a form rendered:
![screenshot 2015-02-19 14 13 06](https://cloud.githubusercontent.com/assets/4390231/6267147/ec8030dc-b841-11e4-8371-5bb72e335fc4.png)
Where `Image ids` field is clickable, but I can't write anything there.

**Comment by maryokhin** [comment]
Update: It's not a bug, but maybe some room for improvement is possible. 
It becomes clickable when at least 1 object you want to submit exists, but before that, very confusing.

![screenshot 2015-02-19 15 24 36](https://cloud.githubusercontent.com/assets/4390231/6268226/8a87582e-b84b-11e4-8ce3-ebc8b72dff66.png)

**Comment by tomchristie** [comment]
Okay. Possible questions:

1. What do we think the display should look like?
2. What does the Django admin do with a similar selection?
3. What do Django forms do with a similar selection?

**Comment by maryokhin** [comment]
Can only answer the first question, because I never really used the other 2 :)

I think the standard practice of graying it out, or any other way of making it visually clear that it's intentional would help.

**Comment by kevin-brown** [comment]
I think the problem here isn't that you can't type anything in - you couldn't in the past - but instead the select box isn't actually rendering all of the possible options.

Of course, we did that in the past and hit major speed issues on the browsable API because `__str__` was being called on every object that was returned. It also opened the doors to data issues, as all users could see all objects unless the queryset was filtered.

> What does the Django admin do with a similar selection? What do Django forms do with a similar selection?

They use a `<select multiple>` box (or the multiselect widget in the admin).

**Comment by tomchristie** [comment]
> instead the select box isn't actually rendering all of the possible options.

Confused? Which options are not being rendered? Surely we should be rendering all items in the queryset, same as we always have and same as eg a django form. Still unclear to me from this convo what, if any, difference there is from if you'd used a form instead of a serializer?

**Comment by kevin-brown** [comment]
> Which options are not being rendered?

My assumption is that `Images.objects.all()` is not a blank queryset, but it looks like (from the image) that none of the options are being rendered. That would at least explain why it just clickable, as that looks like an empty `<select multiple>`.

Though, after looking through the code, it looks like the queryset might actually just be empty. It looks like the `choices` should always be rendered, which would fill up the options and make it clear what the select box is for, but there are none in this case.

**Comment by maryokhin** [comment]
> My assumption is that Images.objects.all() is not a blank queryset
It was an empty queryset on the 1st screenshot, it was that I didn't know what was supposed to be rendered with `PrimaryKeyRelatedField(many=True)` and the fact that it's a big white field, but not selectable that got me confused :)

**Comment by tomchristie** [comment]
Right. So presumably you'd get the same style style on a Django form then, right?

No doubt we *could* enhance this, not clear if we care enough to have it on our radar or not, unless there's something simple, obvious, that we should be doing?

Concrete suggestions on HTML style with associated screenshots for review would be good.

**Comment by kevin-brown** [comment]
> Right. So presumably you'd get the same style style on a Django form then, right?

Correct. Django does not handle this special case at all, but instead displays the empty select like we do.

**Comment by maryokhin** [comment]
But what about some kind of disabled state for situations like these?
http://www.w3schools.com/bootstrap/tryit.asp?filename=trybs_form_horizontal_all&stacked=h

**Comment by tomchristie** [comment]
Yes. Adding the 'disabled' attribute on empty is sufficient. Perhaps do something like this in the template...

    <select disabled multiple class="form-control">
        <option>No items to select</option>
    </select>

Ie. in the template, if choices is empty add the 'disabled' attribute, and a placeholder option.

Renders as:

![screen shot 2015-02-20 at 09 55 46](https://cloud.githubusercontent.com/assets/647359/6284319/a8f726b8-b8e6-11e4-9279-428a91b6ab7d.png)

**Comment by maryokhin** [comment]
:+1: It would make the intent much clearer and overall look cooler.

**Comment by jpadilla** [comment]
Is this best tackled directly on the `select_multiple.html` templates?

**Comment by tomchristie** [comment]
> Is this best tackled directly on the select_multiple.html templates?

Yes.

**Comment by jpadilla** [comment]
Will try this out today

**Comment by tomchristie** [comment]
Closed by #2587.
